compiler/rustc_ty_utils/src/layout.rs

```diff
@@ -498,6 +498,55 @@ fn layout_of_uncached<'tcx>(
                 return Err(error(cx, LayoutError::SizeOverflow(ty)));
             };
 
+            // If the struct tail is sized and can be unsized, check that unsizing doesn't move the fields around.
+            if cfg!(debug_assertions)
+                && maybe_unsized
+                && def
+                    .non_enum_variant()
+                    .fields
+                    .raw
+                    .last()
```

Comment:
this could be extracted as a `last_field` since it's also used in the maybe unsized calculation